File: examples/python/testing-patterns/complex-workflows/long-waits-workflow.py
@durable_execution
def handler(event: dict, context: DurableContext) -> str:
    context.step(lambda _: "Starting", name="start")
    context.wait(Duration.from_seconds(3600), name="long_wait")  # 1 hour
    context.step(lambda _: "Continuing", name="continue")
    return "Complete"

File: examples/java/testing-patterns/complex-workflows/long-waits-workflow.java
// Long waits workflow handler
public class LongWaitHandler extends DurableHandler<Map, String> {
    @Override
    public String handleRequest(Map input, DurableContext ctx) {
        ctx.step("start", String.class, stepCtx -> "Starting");
        ctx.wait("long_wait", Duration.ofHours(1));
        ctx.step("continue", String.class, stepCtx -> "Continuing");
        return "Complete";
    }
}
